compiler/rustc_middle/src/query/mod.rs

```diff
@@ -948,27 +952,37 @@ rustc_queries! {
     }
 
     TypeChecking {
-        query all_local_trait_impls(key: CrateNum) -> &'tcx BTreeMap<DefId, Vec<hir::HirId>> {
+        /// Return all `impl` blocks in the current crate.
+        ///
+        /// To allow caching this between crates, you must pass in [`LOCAL_CRATE`] as the crate number.
+        /// Other crate numbers will return the same results, but will break caching.
+        ///
+        /// [`LOCAL_CRATE`]: rustc_hir::def_id::LOCAL_CRATE
+        query all_local_trait_impls(local_crate: CrateNum) -> &'tcx BTreeMap<DefId, Vec<hir::HirId>> {
```

Comment:
If the crate num is not `LOCAL_CRATE`, the `provide_extern` version of the query code is executed instead of the `provide` version.

Reply:
There is no `provide_extern` version of this function. What happens in that case?

Reply:
An ICE.

Reply:
Thanks. I fixed the comment.